internal/datastore/postgres/watch.go

```diff
@@ -116,7 +114,7 @@ func (pgd *pgDatastore) loadChanges(
 		return
 	}
 
-	stagedChanges := make(map[uint64]*changeRecord)
+	stagedChanges := make(common.Changes)
```

Comment:
Since this is non-obvious, maybe a `common.NewChanges()` that does the `make`? (especially since if we do decide to change the type to a struct down the road, the `make` wouldn't work)